feat(curriculum): adding questions for Forms, Data Fetching and Routing Quiz (#59183)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-react-forms-data-fetching-and-routing/66f1ad049d7a6ac0886cc2ba.md

### --question--

#### --text--

Which of the following hooks is used to memoize computed values?

#### --distractors--

`useParams`

---

`useEffect`

---

`useState`

#### --answer--

`useMemo`

### --question--

#### --text--

Which of the following uses the correct hook to memoize function references?

#### --distractors--

```js
useContext(() => {
 // code goes here
}, [dependency]);
``` 

---

```js
useParams(() => {
 // code goes here
}, [dependency]);
``` 

---

```js
useState(() => {
 // code goes here
}, [dependency]);
``` 

#### --answer--

```js
useCallback(() => {
 // code goes here
}, [dependency]);
``` 

### --question--

#### --text--

What is a dependency?

#### --distractors--

A dependency is a commonly used testing library used in React applications.

---

A dependency is a type of React component that only runs on the server.

---

A dependency is a special compiler used in React applications to compile the React code into JavaScript. 

#### --answer--

A dependency is where one component or module in an application relies on another to function properly.

### --question--

#### --text--

What is a package manager?

#### --distractors--

> A tool used to update only third party packages installed in applications.

Reviewer: ```suggestion
A tool used to update third-party packages installed in applications.
```

---

> A tool used to manage only packages created within the project.

Reviewer: ```suggestion
A tool used to manage packages solely created within a project.
```
